### PR TITLE
Feature/model list filter

### PR DESCRIPTION
**Pull request by JozefFlakus**

### CODE REVIEW

File: src/app/+models/components/model-filter/model-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor(private store: Store<State>) { }
+
+  ngOnInit() {
+    this.filter$
+      .pipe(map(val => val.toLowerCase().trim()))
+      .subscribe(val => this.store.dispatch(new modelActions.ModelSetFilterValue(val)));
+  }
+
+  ngOnDestroy() { }
```
Comment (sebastianmusial) [inline src/app/+models/components/model-filter/model-filter.component.ts]: Can you remove ngOnDestroy?

File: src/app/+models/components/model-table/model-table.component.ts [hunk 2]
```diff
@@ -25,6 +40,8 @@ export class ModelTableComponent implements OnChanges {
     );
   }
 
+  ngOnDestroy() { }
```
Comment (sebastianmusial) [inline src/app/+models/components/model-table/model-table.component.ts]: Can we remove `ngOnDestroy`?

File: src/app/+models/components/model-filter/model-filter.component.ts (hunk 1 above)
Comment (JozefFlakus) [reply src/app/+models/components/model-filter/model-filter.component.ts]: no - it is required by @AutoUnsubscribe() decorator

File: src/app/+models/components/model-table/model-table.component.ts (hunk 2 above)
Comment (JozefFlakus) [reply src/app/+models/components/model-table/model-table.component.ts]: no - it is required by @AutoUnsubscribe() decorator

**Pull request merged**